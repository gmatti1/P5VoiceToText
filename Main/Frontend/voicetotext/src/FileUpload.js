import React from 'react'
import axios, { post } from 'axios';

class FileUpload extends React.Component {

  constructor(props) {
    super(props);
    this.state ={
      file:null
    }
    this.OnSubmittingForm = this.OnSubmittingForm.bind(this)
    this.onChangingForm = this.onChangingForm.bind(this)
    this.Upload_file = this.Upload_file.bind(this)
  }
  Upload_file(file){
    const url = 'http://example.com/file-upload';
    const formData = new FormData();
    formData.append('file',file)
    const config = {
        headers: {
            'content-type': 'multipart/form-data'
        }
    }
    return  post(url, formData,config)
  }
  OnSubmittingForm(e){
    e.preventDefault() 
    this.Upload_file(this.state.file).then((response)=>{
      console.log(response.data);
    })
  }
 
 
  onChangingForm(e) {
    this.setState({file:e.target.files[0]})
  }

  render() {
    return (
      <form onSubmit={this.OnSubmittingForm}>
        <h1>Please upload the audio file</h1>
        <input type="file" onChangingForm={this.onChangingForm} />
        <button type="submit">Upload</button>
      </form>
   )
  }
}



export default FileUpload

// References https://gist.github.com/AshikNesin
//References: react website
