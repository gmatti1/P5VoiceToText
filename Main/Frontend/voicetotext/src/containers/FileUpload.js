import React from 'react';
import './../styles/App.css';
import axios from 'axios';
import hist from './../assets/hist.png';
import { Link } from './../../node_modules/react-scroll';

class FileUpload extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isFileUploaded: null
    };

    this.OnSubmittingForm = this.OnSubmittingForm.bind(this);
    this.Upload_file = this.Upload_file.bind(this);
  }

 Upload_file(file) {
    const formData = new FormData();
    formData.append('file', this.uploadInput.files[0]);
Promise.all([fetch('/uploadVoiceFile', {
  method: 'POST',
  body: formData
}),
fetch(' http://localhost:5000/convertVoice ',{
  method: 'GET',
 
})
]).then(([response1,response2]) => {
  
  response2.json().then(data => {this.setState({ "data": data });
    console.log(this.state.data);
  });
       
    });
  }
  

 


  OnSubmittingForm(e) {
    e.preventDefault();
    this.Upload_file();
  }
  

  render() {
    return (
	<div>
      <form onSubmit={this.OnSubmittingForm}>
        <h1 className="Uploadheader">Please upload the audio file</h1>
        <input className ="Input"
          type='file'
          ref={ref => {
            this.uploadInput = ref;
          }}
          name='voiceFile'
        />
        <button className="Buttonformat" type='submit' >Upload</button>		 		
      </form>
	  <Link
          activeClass='active'
          to='hist'
          spy={true}
          smooth={true}
          offset={0}
          duration={500}>
            <button className="Historybutton" type='submit'><img src={hist} className='Histicon' alt='History' />
				<span className="tooltiptext">History : All the Uploaded files before</span>
			</button>
        </Link>
      </div>
    );
  }
}

export default FileUpload;
