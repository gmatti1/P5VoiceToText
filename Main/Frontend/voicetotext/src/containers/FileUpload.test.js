import React from "react";
import { shallow } from "enzyme";
import FileUpload from "./FileUpload";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });
beforeAll(() => {
    global.fetch = jest.fn();
    //window.fetch = jest.fn(); if running browser environment
  });
  let wrapper;

describe("FileUpload", () => {
    
    let mockSubmit;
    beforeEach(() => {
      mockSubmit = jest.fn();
      wrapper = shallow(<FileUpload submit={mockSubmit} />);
    });
   
    it("should match the snapshot", () => {
        expect(wrapper).toMatchSnapshot();
      });




describe("handleChange", () => {
    it("should call setState on title", () => {
        const mockPreventDefault = jest.fn();
    const mockEvent = {
      preventDefault: mockPreventDefault,
      target: {
            
        value: "test"

    }
    };
      
      const expected = {
        isFileUploaded: null,
      title: '',
      textCategorized: [],
     
      loading: false,
      filename: '',
      
      textdone: false,
      disabled: false,
      istextupdated: false,
      targetElement: null,
      stats: [],
        convertedText:"test"
      };
      wrapper.instance().handleChange(mockEvent);
      
      expect(wrapper.state()).toEqual(expected);
    });
   
  });
  
})
  