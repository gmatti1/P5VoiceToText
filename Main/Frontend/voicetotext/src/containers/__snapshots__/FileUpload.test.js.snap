// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileUpload should match the snapshot 1`] = `ShallowWrapper {}`;
